### ISSUE TITLE
Document migration consolidation

### ISSUE THREAD

**Issue by Bilge** [cakephp/phinx#2008]
Even if there is no formal command or other provision to support consolidation of migration files, it would still be useful to document a working procedure to achieve this. Migration consolidation is the normal process of reducing the number of migration files down to zero, or near-zero periodically, to eliminate old, obsolete migrations and speed up database seeding.

**Comment by broberts-mrmc** [comment]
Along these lines, if the migrations themselves cannot be easily consolidated (or even if they can), maybe a small step would be to have built-in support (i.e. configuration option) for automatically organizing migrations into subdirectories by year or year and month. After a few years, in a long-running or on-going project, as we have, its easy to amass hundreds of migrations, particularly if you subscribe to the idea of limiting the number of database changes performed in each individual migration or, at least, keeping each migration as single-purpose as possible (i.e. there tends to be more migrations when taking this approach). So, in that case, or just after a longer period of time with the "beefy migration" approach, it doesn't seem helpful for most of those migrations to remain in "direct view" in the migrations directory, particularly those created more than 12 months ago.

**Comment by dereuromark** [comment]
The subdirectory approach sounds more like a workaround then, if you have that many, I also agree that just merging them in the end would be probably the best long-term kind of resolution on this.
But it can still be done in parallel of course.

I also have to scroll already multiple pages on some repos to get to the lastest one :)

**Comment by Bilge** [comment]
@broberts-mrmc Migration *organization* should be considered a separate topic from *consolidation*. Please create a separate issue to discuss that.

**Comment by broberts-mrmc** [comment]
I could agree if you weren't iDead. As a general rule, I don't make agreements with the dead. Seriously, as one issue or separate issues, it makes me little difference. Though, I think the topics are related so, if separate issues, a reference between the separate issues would seem appropriate.

**Comment by dereuromark** [comment]
I would like to link here to laravels docs on this https://laravel.com/docs/8.x/migrations#squashing-migrations
As this might clarify some terminology options.

Also, I think there might be already some existing issues or PRs around a similar topic
So far I only found https://github.com/cakephp/phinx/issues/102

**Comment by Bilge** [comment]
I believe Doctrine also has a command called _rollup_, although it seems [scarcely documented](https://symfony.com/doc/current/bundles/DoctrineMigrationsBundle/index.html#usage), so I'm not sure if it is exactly the same thing. There are, however, some recent [blog](https://localheinz.com/blog/2020/06/10/rolling-up-database-migrations-with-doctrine/) [posts](https://odolbeau.fr/blog/doctrine-migrations-rollup.html) on the topic.

**Comment by Bilge** [comment]
So... how do we rollup migrations, guys?
